--- a/C:\AGE\AGE2D\Engine\src\AGE\TileMap\Public\TileMapManager.h
+++ b/C:\AGE\AGE2D\Engine\src\AGE\TileMap\Public\TileMapManager.h
@@ -9,7 +9,12 @@
 	public:
 
 		TileMapManager();
-		~TileMapManager() = default;
+		/**
+ * @brief Default destructor for the TileMapManager class.
+ *
+ * This function is responsible for releasing any resources that were acquired by the TileMapManager during its lifetime, such as memory or file handles. It does not perform any operations on the objects themselves.
+ */
+~TileMapManager() = default;
 
 	private:
 
