--- a/C:\AGE\AGE2D\Engine\src\AGE\Audio\AudioEngine\Private\AudioEngine.cpp
+++ b/C:\AGE\AGE2D\Engine\src\AGE\Audio\AudioEngine\Private\AudioEngine.cpp
@@ -7,7 +7,8 @@
 
 namespace AGE
 {
-    Ref<AudioEngine> AudioEngine::Create(AudioEngineType Type)
+    
+Ref<AudioEngine> AudioEngine::Create(AudioEngineType Type)
     {
 
         switch (Type)
@@ -44,7 +45,11 @@
     }
 
     template<typename T>
-    T* AudioEngine::As()
+    /**
+ * @brief This function is a placeholder and will always fail an assertion. It's used as a stub for future development.
+ * @return Unknown
+ */
+T* AudioEngine::As()
     {
         CoreLogger::Assert(false, "As() Failed");
     }
