--- a/C:\AGE\AGE2D\Engine\src\AGE\Platform\OpenGL\Private\OpenGLFrameBuffer.cpp
+++ b/C:\AGE\AGE2D\Engine\src\AGE\Platform\OpenGL\Private\OpenGLFrameBuffer.cpp
@@ -6,22 +6,57 @@
 {
 	namespace Utils
 	{
-		static GLenum TextureTarget(bool Multisampled)
+		/**
+ * @brief Determines the OpenGL texture target based on whether multisampling is enabled.
+ * 
+ * This function takes a boolean parameter indicating if multisampling should be used. If true, it returns GL_TEXTURE_2D_MULTISAMPLE; otherwise, it returns GL_TEXTURE_2D.
+ * 
+ * @param Multisampled A boolean value indicating whether to use multisampling or not.
+ * @return The OpenGL texture target corresponding to the input parameter.
+ */
+static GLenum TextureTarget(bool Multisampled)
 		{
 			return Multisampled ? GL_TEXTURE_2D_MULTISAMPLE : GL_TEXTURE_2D;
 		}
 	
-		static void CreateTextures(bool Multisampled, uint32_t* OutID, uint32_t Count)
+		/**
+ * @brief Creates a set of textures.
+ *
+ * This function creates a set of OpenGL textures using the glCreateTextures function. The number of textures to be created is specified by 'Count'. 
+ * The parameter 'Multisampled' determines whether the textures are multisampled or not, and 'OutID' is an array that will hold the IDs of the newly created textures.
+ *
+ * @param Multisampled A boolean value indicating if the textures should be multisampled.
+ * @param OutID An array to store the IDs of the newly created textures.
+ * @param Count The number of textures to create.
+ * 
+ * @return void
+ */
+static void CreateTextures(bool Multisampled, uint32_t* OutID, uint32_t Count)
 		{
 			glCreateTextures(TextureTarget(Multisampled), (int)Count, OutID);;
 		}
 	
-		static void BindTexture(bool Multisampled, uint32_t ID)
+		/**
+ *  @brief This function binds a texture to the OpenGL context.
+ *  
+ *  The function takes two parameters, a boolean indicating whether the texture is multisampled and an unsigned integer representing the ID of the texture. It uses these values to determine the appropriate target for binding the texture with glBindTexture().
+ *  
+ *  @param Multisampled A boolean value that indicates if the texture is multisampled or not.
+ *  @param ID An unsigned integer representing the ID of the texture to be bound.
+ *  
+ *  @return void
+ */
+static void BindTexture(bool Multisampled, uint32_t ID)
 		{
 			glBindTexture(TextureTarget(Multisampled), ID);
 		}
 	
-		static void AttachColorTexture(uint32_t ID, int Samples, GLenum InternalFormat, GLenum Format, uint32_t Width, uint32_t Height, int Index)
+		/**
+ * @brief Attaches a color texture to the framebuffer.
+ * 
+ * This function sets up and attaches a color texture to the OpenGL framebuffer object. The texture can be either multisampled or regular, depending on the 'Samples' parameter. It then binds this texture as a color attachment point for rendering.
+ */
+static void AttachColorTexture(uint32_t ID, int Samples, GLenum InternalFormat, GLenum Format, uint32_t Width, uint32_t Height, int Index)
 		{
 			bool Multisampled = Samples > 1;
 	
@@ -44,7 +79,8 @@
 		}
 	
 	
-		static void AttachDepthTexture(uint32_t ID, int Samples, GLenum Format, GLenum AttachmentType, uint32_t Width, uint32_t Height)
+		
+static void AttachDepthTexture(uint32_t ID, int Samples, GLenum Format, GLenum AttachmentType, uint32_t Width, uint32_t Height)
 		{
 			bool Multisampled = Samples > 1;
 	
@@ -67,7 +103,15 @@
 		}
 	
 	
-		static bool IsDepthFormat(FramebufferTextureFormat Format)
+		/**
+ * @brief Checks if the given format is a depth format.
+ * 
+ * This function checks whether the provided FramebufferTextureFormat is one of the depth formats. It returns true for DEPTH24STENCIL8 and false otherwise.
+ * 
+ * @param Format The format to check.
+ * @return True if the format is a depth format, false otherwise.
+ */
+static bool IsDepthFormat(FramebufferTextureFormat Format)
 		{
 			switch (Format)
 			{
@@ -85,7 +129,15 @@
 	
 		}
 
-		static GLenum AGETextureFormatToGL(FramebufferTextureFormat Format)
+		/**
+ * @brief Converts a FramebufferTextureFormat to its corresponding GLenum.
+ *
+ * This function takes in a FramebufferTextureFormat and returns the equivalent GLenum value. It handles four formats: RGBA8, RED_INTEGER. For any other format, it asserts false and returns 0.
+ *
+ * @param Format The FramebufferTextureFormat to convert.
+ * @return The corresponding GLenum for the input FramebufferTextureFormat. If the input is not a recognized format, it will return 0.
+ */
+static GLenum AGETextureFormatToGL(FramebufferTextureFormat Format)
 		{
 			switch (Format)
 			{
@@ -109,7 +161,16 @@
 	
 		static const uint32_t s_MaxFramebufferSize = 8192;
 	
-		OpenGLFrameBuffer::OpenGLFrameBuffer(const FrameBufferSpecification& Spec)
+		/**
+ * @brief Constructor for OpenGLFrameBuffer class.
+ *
+ * This constructor takes a FrameBufferSpecification object as input, which contains the specifications of the frame buffer to be created. 
+ * It iterates over the Attachments in the specification and separates them into color attachments (those that are not depth formats) and depth attachment (the one that is).
+ * Finally, it calls Invalidate() function to create the actual frame buffer object with OpenGL.
+ *
+ * @param Spec The FrameBufferSpecification object containing specifications for the frame buffer.
+ */
+OpenGLFrameBuffer::OpenGLFrameBuffer(const FrameBufferSpecification& Spec)
 			:m_Specification(Spec)
 		{
 			for (auto FBSpec : m_Specification.Attachments.Attachments)
@@ -126,13 +187,18 @@
 
 			Invalidate();
 		}
-		OpenGLFrameBuffer::~OpenGLFrameBuffer()
+		/**
+ * @brief Destructor for the OpenGLFrameBuffer class.
+ * This function deletes both color and depth attachments associated with the frame buffer object.
+ */
+OpenGLFrameBuffer::~OpenGLFrameBuffer()
 		{
 			glDeleteFramebuffers(1, &m_RendererID);
 			glDeleteTextures((int)m_ColorAttachments.size(), m_ColorAttachments.data());
 			glDeleteTextures(1, &m_DepthAttachment);
 		}
-		void OpenGLFrameBuffer::Invalidate()
+		
+void OpenGLFrameBuffer::Invalidate()
 		{
 	
 			if (m_RendererID)
@@ -214,7 +280,16 @@
 			CoreLogger::Assert(glCheckFramebufferStatus(GL_FRAMEBUFFER) == GL_FRAMEBUFFER_COMPLETE, "Framebuffer is incomplete");
 			glBindFramebuffer(GL_FRAMEBUFFER, 0);
 		}
-		void OpenGLFrameBuffer::Resize(const uint32_t Width, const uint32_t Height)
+		/**
+ * @brief Resizes the OpenGL FrameBuffer to a new size.
+ *
+ * This function resizes the framebuffer to the specified width and height, provided they are within the maximum allowed size (s_MaxFramebufferSize). 
+ * If the dimensions are invalid or exceed the maximum size, it logs a warning message and returns without changing anything.
+ *
+ * @param Width The new width for the FrameBuffer.
+ * @param Height The new height for the FrameBuffer.
+ */
+void OpenGLFrameBuffer::Resize(const uint32_t Width, const uint32_t Height)
 		{
 			if (Width == 0 || Height == 0 || Width > s_MaxFramebufferSize || Height > s_MaxFramebufferSize)
 			{
@@ -227,19 +302,45 @@
 	
 			Invalidate();
 		}
-		void OpenGLFrameBuffer::Bind()
+		/**
+ * @brief This function binds the OpenGL framebuffer to be used for rendering operations.
+ * 
+ * The function first calls glBindFramebuffer with GL_FRAMEBUFFER and m_RendererID as arguments, which binds the framebuffer object with the given ID.
+ * Then it sets the viewport using glViewport with (0, 0) and (m_Specification.Width, m_Specification.Height) as arguments, setting the dimensions of the rendering area to match the size of the framebuffer's client area.
+ */
+void OpenGLFrameBuffer::Bind()
 		{
 			glBindFramebuffer(GL_FRAMEBUFFER, m_RendererID);
 	
 			glViewport(0, 0, (int)m_Specification.Width, (int)m_Specification.Height);
 		}
-		void OpenGLFrameBuffer::Unbind()
+		/**
+ * @brief Unbinds the current frame buffer.
+ *
+ * This function binds the default frame buffer to the OpenGL context using glBindFramebuffer().
+ * The GL_FRAMEBUFFER target is used, and 0 is passed as argument to unbind the currently bound frame buffer.
+ *
+ * @return void
+ */
+void OpenGLFrameBuffer::Unbind()
 		{
 			glBindFramebuffer(GL_FRAMEBUFFER, 0);
 			
 		}
 
-		int OpenGLFrameBuffer::ReadPixel(uint32_t AttachmentIndex, int x, int y)
+		/**
+ * @brief Reads a single pixel from the OpenGL framebuffer.
+ *
+ * This function reads a single pixel at the specified (x, y) coordinates in the color attachment of the framebuffer. 
+ * The pixel data is returned as an integer value.
+ *
+ * @param AttachmentIndex Index of the color attachment to read from. It should be less than the number of elements in m_ColorAttachments array.
+ * @param x X-coordinate of the pixel to read.
+ * @param y Y-coordinate of the pixel to read.
+ * 
+ * @return The integer value of the pixel at (x, y) coordinates in the specified color attachment.
+ */
+int OpenGLFrameBuffer::ReadPixel(uint32_t AttachmentIndex, int x, int y)
 		{
 			CoreLogger::Assert(AttachmentIndex < m_ColorAttachments.size(), "Attachment Index is larger than number of Elements in Color Attachments array!");
 
@@ -250,7 +351,17 @@
 			return PixelData;
 		}
 
-		void OpenGLFrameBuffer::ClearAttachment(uint32_t Index , int Value)
+		/**
+ * @brief Clears a specific color attachment at the given index with a specified value.
+ *
+ * This function clears the color attachment at the provided index to the specified value using OpenGL's glClearTexImage function. 
+ * It first checks if the provided index is within the valid range of color attachments, and then retrieves the corresponding specification for that attachment.
+ * The texture format from the specification is used in the call to glClearTexImage. If the index is out of bounds or there are no color attachments at all, it will assert with an error message. 
+ *
+ * @param Index The zero-based index of the color attachment to clear.
+ * @param Value The value to which to clear the specified color attachment.
+ */
+void OpenGLFrameBuffer::ClearAttachment(uint32_t Index , int Value)
 		{
 
 			CoreLogger::Assert(Index < m_ColorAttachments.size(), "Index is greater that number of color attachments!");
@@ -259,7 +370,14 @@
 			glClearTexImage(m_ColorAttachments[Index], 0, Utils::AGETextureFormatToGL(Spec.TextureFormat), GL_INT, &Value);
 		}
 
-		void OpenGLFrameBuffer::OnEvent(Event& E)
+		/**
+ * @brief Handles events for the OpenGLFrameBuffer class.
+ * 
+ * This function takes an event as input and processes it based on its type. The exact behavior of this function depends on the specifics of the Event class, which is not defined in this file.
+ * 
+ * @param E Reference to the Event object that needs processing.
+ */
+void OpenGLFrameBuffer::OnEvent(Event& E)
 		{
 		}
 	
