--- a/C:\AGE\AGE2D\Engine\src\AGE\UI\Components\Private\UiImageComponent.cpp
+++ b/C:\AGE\AGE2D\Engine\src\AGE\UI\Components\Private\UiImageComponent.cpp
@@ -14,7 +14,16 @@
 }
 namespace AGE
 {
-	UIImageComponent::UIImageComponent(const std::string &Name)
+	/**
+ * @brief Constructs an instance of UIImageComponent with the given name and initializes it.
+ * 
+ * This constructor creates a new UIImageComponent with the provided name, sets its type to ImageComponent,
+ * retrieves all textures from the AssetManager's asset registry, stores their names in m_TextureNames,
+ * and finally creates an instance of Texture2D using default parameters.
+ * 
+ * @param Name The name for this UIImageComponent.
+ */
+UIImageComponent::UIImageComponent(const std::string &Name)
 	{
 		m_Name = Name;
 		m_Type = UIComponentType::ImageComponent;
@@ -28,16 +37,34 @@
 		m_Image = Texture2D::Create(TextureSpecification());
 	}
 
-	void UIImageComponent::OnUpdate(TimeStep DeltaTime)
+	/**
+ * @brief This function updates the image component.
+ * 
+ * The function takes a TimeStep parameter representing the time elapsed since the last frame. It uses this value to update the image and its properties.
+ * 
+ * @param DeltaTime - A TimeStep object representing the time elapsed since the last frame.
+ * 
+ * @return void
+ */
+void UIImageComponent::OnUpdate(TimeStep DeltaTime)
 	{
 		Renderer2D::DrawQuad(m_Image, m_Properties);
 	}
 
-	void UIImageComponent::OnEvent(Event &Event)
+	/**
+ * @brief Handles an event of type Event.
+ *
+ * This function is responsible for handling events dispatched by the system. It takes in a reference to an Event object and processes it accordingly. 
+ * The exact behavior depends on the specifics of the Event subclass that is being handled.
+ *
+ * @param[in] Event - A reference to the event to be processed.
+ */
+void UIImageComponent::OnEvent(Event &Event)
 	{
 	}
 
-	void UIImageComponent::DrawContent()
+	
+void UIImageComponent::DrawContent()
 	{
 		//Combo Box
 
