--- a/C:\AGE\AGE2D\Engine\src\AGE\Project\Private\Project.cpp
+++ b/C:\AGE\AGE2D\Engine\src\AGE\Project\Private\Project.cpp
@@ -10,7 +10,17 @@
 	std::array<std::string, 14> Project::m_DirectoryNames = { "/Assets", "/Assets/Quests", "/Assets/InventoryDatabase", "/Assets/VisualScripting", "/Scenes", "/Shaders", "/Saves","/Config", "/src","/src/Base","/src/Base/Private","/src/Base/Public","/src/UI/Private","/src/UI/Public"};
 	std::array<std::string, 5> Project::m_GameContDirNames = { "/Textures", "Sounds/Banks", "/Aesprite", "/Shaders", "/UI"};
 
-	void Project::WriteProjectConfig(const std::filesystem::path& Path, const std::string& ProjectName)
+	/**
+ * @brief This function writes project configuration to an INI file.
+ * 
+ * The function takes two parameters, a const reference to a std::filesystem::path object and a string representing the name of the project. It then creates an instance of IniWriter with the path to the ProjectConfig.ini file as its argument. This writer is used to write four key-value pairs into the INI file: "GameContentPath", "GameSourcePath", "GameShadersPath" and "GameScenesPath". The values for these keys are formatted strings that represent paths relative to the project directory, using the provided ProjectName as a variable.
+ * 
+ * @param Path A const reference to a std::filesystem::path object representing the path to the project directory.
+ * @param ProjectName A string representing the name of the project.
+ * 
+ * @return void
+ */
+void Project::WriteProjectConfig(const std::filesystem::path& Path, const std::string& ProjectName)
 	{
 		IniWriter Writer(Path.string() + "/ProjectConfig.ini");
 		Writer.Write("Paths", "GameContentPath", std::vformat("/{}/GameContent/Assets/", std::make_format_args(ProjectName)));
@@ -21,7 +31,18 @@
 
 	}
 
-	void Project::WriteEditorConfig(const std::filesystem::path& Path, const std::string& ProjectName)
+	/**
+ * @brief This function writes an EditorConfig file for a project at the specified path.
+ * 
+ * The function takes two parameters, `Path` and `ProjectName`. It generates an INI configuration file named "EditorConfig.ini" in the directory pointed to by `Path`.
+ * The generated config includes paths for Logs ("/{ProjectName}/Logs/") and Editor Assets ("//{CWD}/Assets/"). 
+ * Here, {ProjectName} is replaced with the actual project name provided as an argument and {CWD} stands for the current working directory.
+ * The function does not return anything, so it has a void return type.
+ * 
+ * @param Path The path where the EditorConfig file will be created.
+ * @param ProjectName The name of the project to use in generating paths.
+ */
+void Project::WriteEditorConfig(const std::filesystem::path& Path, const std::string& ProjectName)
 	{
 		std::string CWD = std::filesystem::current_path().string();
 
@@ -31,7 +52,14 @@
 		Writer.SaveFile();
 	}
 
-	void Project::ReadProjectConfig(const std::filesystem::path &Path)
+	/**
+ * @brief Reads the project configuration from an INI file and updates the AppConfig object accordingly.
+ * 
+ * This function reads a set of paths for various parts of the game project, such as content, source code, shaders, and scenes. It uses an `IniReader` to read the values from the specified path in the filesystem. The configuration data is then used to update the AppConfig object. If multiple entries exist for a particular key (e.g., "GameContentPath"), all of them are stored as a vector of strings.
+ * 
+ * @param Path The path to the INI file containing the project configuration data.
+ */
+void Project::ReadProjectConfig(const std::filesystem::path &Path)
 	{
 		AppConfig& Config = App::Get().GetAppConfig();
 		IniReader Reader(Path.string() + "/ProjectConfig.ini");
@@ -78,7 +106,12 @@
 
 	}
 
-	void Project::ReadEditorConfig(const std::filesystem::path &Path)
+	/**
+ * @brief Reads editor configuration from an INI file.
+ * @param Path The path to the INI file.
+ * @return void
+ */
+void Project::ReadEditorConfig(const std::filesystem::path &Path)
 	{
 		AppConfig& Config = App::Get().GetAppConfig();
 		IniReader Reader(Path.string() + "/EditorConfig.ini");
@@ -107,7 +140,8 @@
 
 	}
 
-	void Project::AddBuiltScenes()
+	
+void Project::AddBuiltScenes()
 	{
 		AppConfig Config = App::Get().GetAppConfig();
 		if (std::filesystem::is_directory(Config.CurrentProjectPath.string() + "/"+s_ActiveProject->GetConfig().Name + "/BuiltScenes/"))
@@ -151,7 +185,8 @@
 
 	}
 
-	Ref<Project> AGE::Project::New(const std::string& ProjectName)
+	
+Ref<Project> AGE::Project::New(const std::string& ProjectName)
 	{
 		AppConfig Config = App::Get().GetAppConfig();
 		if (ProjectName == "Untitled")
@@ -184,7 +219,8 @@
 		}
 		return s_ActiveProject;
 	}
-	Ref<Project> Project::Load(const std::filesystem::path& Path)
+	
+Ref<Project> Project::Load(const std::filesystem::path& Path)
 	{
 		AppConfig& Config = App::Get().GetAppConfig();
 		if (Path.string().find("GAMENAMEHERE",0) != std::string::npos)
@@ -206,7 +242,21 @@
 		CoreLogger::Info("Could not Load Project!\n\tProject File Path {0}", Path.string());
 		return nullptr;
 	}
-	bool Project::SaveActive(const std::filesystem::path& Path, uint16_t AudioEngine, int Renderer, const std::filesystem::path& ScenePath, const std::filesystem::path& QuestPath, const std::filesystem::path& ConfigPath)
+	/**
+ * @brief Saves the active project to a specified path with various configuration options.
+ * 
+ * This function serializes the current active project into a file at the given path, along with additional configurations such as audio engine, renderer, start scene, quest filepath and config filepath. It also adds built-in scenes if they haven't been added before.
+ * 
+ * @param Path The path where to save the project.
+ * @param AudioEngine The audio engine to be used for rendering audio in the project.
+ * @param Renderer The renderer to be used for rendering graphics in the project.
+ * @param ScenePath The path of the start scene file.
+ * @param QuestPath The path of the quest file.
+ * @param ConfigPath The path of the config file.
+ * 
+ * @return Returns true if the serialization was successful, false otherwise.
+ */
+bool Project::SaveActive(const std::filesystem::path& Path, uint16_t AudioEngine, int Renderer, const std::filesystem::path& ScenePath, const std::filesystem::path& QuestPath, const std::filesystem::path& ConfigPath)
 	{
 		ProjectSerializer Serializer(s_ActiveProject);
 		s_ActiveProject->m_ProjectDirectory = Path;
@@ -231,7 +281,8 @@
 		}
 		return false;
 	}
-	bool Project::Package(const std::filesystem::path& Path, int TargetPlatform)
+	
+bool Project::Package(const std::filesystem::path& Path, int TargetPlatform)
 	{
 		switch (TargetPlatform)
 		{
@@ -263,7 +314,8 @@
 		}
 		return false;
 	}
-	void Project::CompileProject()
+	
+void Project::CompileProject()
 	{
 		AppConfig Config = App::Get().GetAppConfig();
 #ifdef AG_PLATFORM_WINDOWS
