--- a/C:\AGE\AGE2D\Engine\src\AGE\Physics\Public\Physics2D.h
+++ b/C:\AGE\AGE2D\Engine\src\AGE\Physics\Public\Physics2D.h
@@ -22,9 +22,22 @@
 	{
 	public:
 
-		Physics2D() = default;
+		Ph/**
+ * @brief Default constructor for the ysics2D class.
+ *
+ * This function initializes a new instance of the ysics2D class with default values.
+ * It does not take any parameters and returns an object of type ysics2D.
+ * 
+ * @return A new instance of the ysics2D class with default values.
+ */
+ysics2D() = default;
 
-		virtual ~Physics2D() = default;
+		vi/**
+ * @brief Virtual destructor for the Physics2D class.
+ *
+ * This function is responsible for freeing any resources that were allocated by the object during its lifetime. It does not return anything and thus, it has a void return type.
+ */
+rtual ~Physics2D() = default;
 
 		bool CreateNewPhysicsWorld(Ref<Scene> scene);
 
@@ -68,10 +81,25 @@
 
 		bool QueryHit(const QueryParams& Params);
 
-		Ref<World>& GetWorld() { return m_World; }
+		Re/**
+ * @brief Returns a reference to the World object.
+ *
+ * This function returns a reference to the World object that is currently being used by the program. The returned object can be modified or accessed directly, allowing for changes to be made to the state of the world.
+ * 
+ * @return A reference to the current World object.
+ */
+f<World>& GetWorld() { return m_World; }
 
 		template<typename T>
-		static T GetShapeFromID(b2ShapeId ID)
+		st/**
+ * @brief Retrieves a shape from its ID.
+ * 
+ * This function retrieves and returns the shape based on the provided ID. The type of shape returned depends on the value of the ID, which is determined by calling b2Shape_GetType().
+ * 
+ * @param[in] ID The unique identifier for a shape.
+ * @return A shape object corresponding to the given ID. If no matching shape exists, it returns a default polygon shape.
+ */
+atic T GetShapeFromID(b2ShapeId ID)
 		{
 			switch (b2Shape_GetType(ID))
 			{
