--- a/C:\AGE\AGE2D\Engine\src\AGE\Physics\Public\World2D.h
+++ b/C:\AGE\AGE2D\Engine\src\AGE\Physics\Public\World2D.h
@@ -24,7 +24,11 @@
         virtual void QuerySegmentOverlap(const QueryParams& Params) override;
         virtual void QueryHit(const QueryParams& Params) override;
 
-        b2QueryFilter& GetQueryFilter() { return m_QueryFilter; }
+        /**
+ * @brief Returns a reference to the query filter object used by this class.
+ * @return A reference to the b2QueryFilter object.
+ */
+b2QueryFilter& GetQueryFilter() { return m_QueryFilter; }
 
         b2BodyDef MakeBodyDefinition(const BodyType& Type, const Vector3& Translation, const Vector3 Rotation, bool IsRotationFixed, void* UserData);
 
