--- a/C:\AGE\AGE2D\Engine\src\AGE\Parser\Private\JsonParser.cpp
+++ b/C:\AGE\AGE2D\Engine\src\AGE\Parser\Private\JsonParser.cpp
@@ -5,13 +5,20 @@
 #include "GameStructs/Public/GameStructs.h"
 namespace AGE
 {
-	JsonParser::JsonParser(const std::string& FilePath)
-	{
-
-	}
-
-	template<>
-	bool JsonParser::SaveJsonFile<Ref<GameFramework::QuestInfo>>(const std::filesystem::path& Filepath, std::vector<Ref<GameFramework::QuestInfo>>& Data)
+	/**
+ * @brief Constructor for JsonParser class. 
+ * Initializes the object with a JSON file path to parse.
+ *
+ * @param FilePath The path of the JSON file to be parsed.
+ */
+JsonParser::JsonParser(const std::string& FilePath)
+	{
+
+	}
+
+	template<>
+	
+bool JsonParser::SaveJsonFile<Ref<GameFramework::QuestInfo>>(const std::filesystem::path& Filepath, std::vector<Ref<GameFramework::QuestInfo>>& Data)
 	{
 		std::ofstream Out(Filepath);
 		if (Data.empty())
@@ -45,7 +52,8 @@
 	}
 
 	template<>
-	bool JsonParser::SaveJsonFile<GameFramework::ItemInfo>(const std::filesystem::path& Filepath, std::vector<GameFramework::ItemInfo>& Data)
+	"/**\n * @brief This function saves a vector of ItemInfo objects to a JSON file.\n * \n * @param Filepath The path where the JSON file will be saved.\n * @param Data A reference to the vector of ItemInfo objects that will be serialized and written to the file.\n *\n * @return Returns true if successful, false otherwise. If the data vector is empty, this function returns false without attempting to write anything.\n */"
+bool JsonParser::SaveJsonFile<GameFramework::ItemInfo>(const std::filesystem::path& Filepath, std::vector<GameFramework::ItemInfo>& Data)
 	{
 		std::ofstream Out(Filepath);
 		if (Data.empty())
@@ -80,7 +88,16 @@
 	}
 
 	template<>
-	bool JsonParser::SaveJsonFile<GameFramework::PlayerStats>(const std::filesystem::path& Filepath, std::vector<GameFramework::PlayerStats>& Data)
+	/**
+ * @brief This function saves a vector of PlayerStats objects to a JSON file.
+ * 
+ * @param Filepath The path to the JSON file where the data will be saved.
+ * @param Data A reference to a vector containing the PlayerStats objects that will be serialized and written to the file.
+ * @return bool Returns true if successful, false otherwise.
+ * 
+ * @note If the input vector is empty, the function returns false without attempting to write anything to the file.
+ */
+bool JsonParser::SaveJsonFile<GameFramework::PlayerStats>(const std::filesystem::path& Filepath, std::vector<GameFramework::PlayerStats>& Data)
 	{
 		std::ofstream Out(Filepath);
 		if (Data.empty())
@@ -112,7 +129,15 @@
 	}
 
 	template<>
-	bool JsonParser::SaveJsonFile<GameFramework::EnemyStats>(const std::filesystem::path& Filepath, std::vector<GameFramework::EnemyStats>& Data)
+	/**
+ * This function saves a vector of EnemyStats objects to a JSON file at the specified path. The JSON file will contain an object for each EnemyStat in the vector, with keys corresponding to the members of the EnemyStat class and values being the respective member's value. If the directory for the given filepath does not exist, it is created.
+ * 
+ * @param Filepath The path where the JSON file will be saved.
+ * @param Data A reference to the vector of EnemyStats objects that will be saved.
+ * 
+ * @return Returns true if successful, false otherwise. If the data is empty (i.e., no stats to save), it returns false.
+ */
+bool JsonParser::SaveJsonFile<GameFramework::EnemyStats>(const std::filesystem::path& Filepath, std::vector<GameFramework::EnemyStats>& Data)
 	{
 		if (!std::filesystem::exists(Filepath.parent_path()))
 		{
@@ -148,7 +173,15 @@
 	}
 
 
-	nlohmann::json JsonParser::LoadJsonFile(const std::filesystem::path& Filepath)
+	/**
+ * @brief Loads a JSON file from the given path.
+ * 
+ * This function attempts to load a JSON file at the provided path and returns its content as a nlohmann::json object. If the file does not exist, it logs an error message and returns an empty json object. If the file is empty or corrupted, it also logs a warning message and returns an empty json object.
+ * 
+ * @param Filepath The path to the JSON file to load.
+ * @return nlohmann::json An object containing the content of the loaded JSON file. Returns an empty json object if the file does not exist or is empty/corrupted.
+ */
+nlohmann::json JsonParser::LoadJsonFile(const std::filesystem::path& Filepath)
 	{
 		if (!std::filesystem::exists(Filepath))
 		{
@@ -173,7 +206,12 @@
 		nlohmann::json Data = nlohmann::json::parse(In);
 		return Data;
 	}
-	std::string JsonParser::Parse(const std::string& FilePath)
+	/**
+ * @brief Parses a JSON file and returns its content as a string.
+ * @param FilePath The path to the JSON file that should be parsed.
+ * @return A string containing the contents of the JSON file, or an empty string if the file could not be opened or parsed.
+ */
+std::string JsonParser::Parse(const std::string& FilePath)
 	{
 		std::ifstream In(FilePath);
 		nlohmann::json data = nlohmann::json::parse(In);
@@ -182,7 +220,16 @@
 		return Dump;
 
 	}
-	std::string JsonParser::ParseString(const std::string& String)
+	/**
+ * @brief Parses a JSON string and returns its array representation.
+ *
+ * This function takes in a JSON string, parses it using the nlohmann::json library, 
+ * converts it to an array (if possible), and then returns this array as a string.
+ *
+ * @param String The input JSON string to be parsed.
+ * @return A string representation of the parsed JSON data's array. If the JSON data is not an array, "Unknown" will be returned.
+ */
+std::string JsonParser::ParseString(const std::string& String)
 	{
 		nlohmann::json data = nlohmann::json::parse(String);
 
