--- a/C:\AGE\AGE2D\Engine\src\AGE\Core\Public\LayerStack.h
+++ b/C:\AGE\AGE2D\Engine\src\AGE\Core\Public\LayerStack.h
@@ -6,7 +6,8 @@
 
 namespace AGE
 {
-	class AGE_API LayerStack
+	
+class AGE_API LayerStack
 	{
 	public:
 		LayerStack();
@@ -23,13 +24,33 @@
 		Layer* GetLayerByName(const std::string& LayerName);
 
 
-		std::vector<Layer*>::iterator begin() { return m_Layers.begin(); }
+		/**
+ * @brief Returns an iterator pointing to the beginning of the Layers vector.
+ * @return An iterator pointing to the start of the Layers vector. If no layers exist, returns an iterator equal to end().
+ */
+std::vector<Layer*>::iterator begin() { return m_Layers.begin(); }
 
-		std::vector<Layer*>::iterator end() { return m_Layers.end(); }
+		/**
+ * @brief Returns an iterator pointing to the theoretical element that follows the last element of the vector.
+ * @return An iterator to the theoretical element following the last element of the vector.
+ */
+std::vector<Layer*>::iterator end() { return m_Layers.end(); }
 
-		std::vector<Layer*>::const_iterator begin() const { return m_Layers.cbegin(); }
+		/**
+ * @brief Returns a constant iterator pointing to the beginning of the layers vector.
+ * @return A constant iterator to the start of the layers vector.
+ */
+std::vector<Layer*>::const_iterator begin() const { return m_Layers.cbegin(); }
 
-		std::vector<Layer*>::const_iterator end() const { return m_Layers.cend(); }
+		/**
+ * @brief Returns a constant iterator pointing to the past-the-end element of the layer vector.
+ * 
+ * This function returns a constant iterator that points to one position past the last element in the layer vector.
+ * It is used to indicate the end of the sequence of elements in the container, similar to how std::vector's member functions cend() return an iterator pointing to the past-the-end element.
+ * 
+ * @return A constant iterator pointing to the past-the-end element.
+ */
+std::vector<Layer*>::const_iterator end() const { return m_Layers.cend(); }
 
 	private:
 		std::vector<Layer*> m_Layers;
