--- a/C:\AGE\AGE2D\Engine\src\AGE\Core\Public\Log.h
+++ b/C:\AGE\AGE2D\Engine\src\AGE\Core\Public\Log.h
@@ -35,11 +35,34 @@
 
 		static void Init();
 
-		inline static Ref<spdlog::logger>& GetCoreLogger() { return s_AGECoreLogger; }
-		inline static Ref<spdlog::logger>& GetGameLogger() { return s_AGEGameLogger; }
-		inline static std::vector<char>& GetLogs() { return s_Logs; }
-		inline static std::vector<size_t>& GetOffsets() { return s_Offsets; }
-		inline static std::vector<LogType>& GetTypes() { return s_Type; }
+		/**
+ * @brief This function returns a reference to the core logger used by the application.
+ * @return A reference to the core logger.
+ */
+inline static Ref<spdlog::logger>& GetCoreLogger() { return s_AGECoreLogger; }
+		/**
+ * @brief This function returns a reference to the game logger object.
+ *
+ * The game logger is used for logging messages related to the gameplay. It provides detailed information about what's happening in the game, which can be useful for debugging and performance tuning.
+ *
+ * @return A reference to the game logger object.
+ */
+inline static Ref<spdlog::logger>& GetGameLogger() { return s_AGEGameLogger; }
+		/**
+ * @brief This function returns a reference to the vector 's_Logs'. The purpose of this function is to provide access to the logs for other parts of the program.
+ * @return A reference to the vector 's_Logs'
+ */
+inline static std::vector<char>& GetLogs() { return s_Logs; }
+		/**
+ * @brief This function returns a reference to the vector `s_Offsets` which is used for storing offset values.
+ * @return A reference to the vector `s_Offsets`.
+ */
+inline static std::vector<size_t>& GetOffsets() { return s_Offsets; }
+		/**
+ * @brief Returns a reference to the internal vector of LogTypes.
+ * @return Reference to the internal vector of LogTypes.
+ */
+inline static std::vector<LogType>& GetTypes() { return s_Type; }
 
 
 	private:
@@ -56,7 +79,16 @@
 	namespace CoreLogger
 	{
 		template<typename ... Args>
-		void Trace(std::string_view fmt, Args&& ... args)
+		/**
+ * @brief This function logs a trace message with the specified format and arguments. 
+ * The formatted string is prefixed with "[AGECORE] ". It also adds each character of the log line to Log::GetLogs() and updates Log::GetOffsets().
+ * 
+ * @param fmt A std::string_view representing the format string for the trace message.
+ * @param args Variadic arguments representing the values to be formatted into the trace message.
+ * 
+ * @return void
+ */
+void Trace(std::string_view fmt, Args&& ... args)
 		{
 			std::string Line = "[AGECORE] " + std::vformat(fmt, std::make_format_args(args...)) + "\n";
 			size_t Size = Line.size();
@@ -75,7 +107,8 @@
 #endif
 		}
 		template<typename ... Args>
-		void Info(std::string_view fmt, Args&& ... args)
+		
+void Info(std::string_view fmt, Args&& ... args)
 		{
 			std::string Line = "[AGECORE] " + std::vformat(fmt, std::make_format_args(args...)) + "\n";
 			size_t Size = Line.size();
@@ -96,7 +129,15 @@
 #endif
 		}
 		template<typename ... Args>
-		void Warn(std::string_view fmt, Args&& ... args)
+		/**
+ * @brief This function logs a warning message to the console and stores it in the log buffer for later retrieval.
+ * 
+ * @param fmt The format string used to create the warning message. It should follow the same syntax as printf or std::format.
+ * @param args The arguments that will be substituted into the format string.
+ * 
+ * @return None
+ */
+void Warn(std::string_view fmt, Args&& ... args)
 		{
 			std::string Line = "[AGECORE] " + std::vformat(fmt, std::make_format_args(args...)) + "\n";
 			size_t Size = Line.size();
@@ -117,7 +158,16 @@
 #endif
 		}
 		template<typename ... Args>
-		void Error(std::string_view fmt, Args&& ... args)
+		/**
+ * @brief This function logs an error message to the core logger and various other logging systems. 
+ * The error message is formatted using a variadic template, allowing for flexible formatting of the error message.
+ *
+ * @param fmt A string_view representing the format string for the error message.
+ * @param args Variadic arguments representing the values to be inserted into the format string.
+ * 
+ * @return void This function does not return a value.
+ */
+void Error(std::string_view fmt, Args&& ... args)
 		{
 			std::string Line = "[AGECORE] " + std::vformat(fmt, std::make_format_args(args...)) + "\n";
 			size_t Size = Line.size();
@@ -138,7 +188,15 @@
 #endif
 		}
 		template<typename ... Args>
-		void Critical(std::string_view fmt, Args&& ... args)
+		/**
+ * @brief This function logs a critical message to the system. The message is formatted using fmt::format and includes the prefix "[AGECORE] ". 
+ *        It also pushes each character of the logged string into Log::GetLogs() for later analysis, increments the offset by the size of the logged string, 
+ *        and sets the log type to Critical. The function supports both Windows and Linux platforms via preprocessor directives.
+ * @param fmt A std::string_view representing the format string for the message.
+ * @param args Variadic arguments representing the values to be formatted into the message.
+ * @return void
+ */
+void Critical(std::string_view fmt, Args&& ... args)
 		{
 			std::string Line = "[AGECORE] " + std::vformat(fmt, std::make_format_args(args...)) + "\n";
 			size_t Size = Line.size();
@@ -160,7 +218,8 @@
 		}
 #ifdef AGE_ENABLE_ASSERTS
 		template<typename ... Args> // CoreLogger->Assert(true ==  false, "True does not equal false")
-		void Assert(bool Condition, std::string_view fmt, Args&& ... args)
+		"This function is a critical part of debugging and should only be used when you know what you're doing."
+void Assert(bool Condition, std::string_view fmt, Args&& ... args)
 		{
 			if (!(Condition)) {
 				std::string Line = "[AGECOREASSERT] " + std::vformat(fmt, std::make_format_args(args...)) + "\n";
@@ -190,6 +249,13 @@
 		}
 #else// Just Do Nothing
 		template<typename ... Args>
+/**
+ * @brief This function is used for asserting conditions in the code. If the condition is false, it throws an exception with a formatted message.
+ * @param Condition The condition to be checked. It should evaluate to true if the program state is valid and false otherwise.
+ * @param fmt A format string that describes how the arguments should be formatted when included into the error message.
+ * @param args Variadic arguments representing values to include in the error message.
+ * @return void This function does not return a value. It throws an exception if the condition is false.
+ */
 void Assert(bool Condition, std::string_view fmt, Args&& ... args){}
 #endif
 
@@ -199,7 +265,17 @@
 	{
 		// GameLogger::Trace("This is a logged event {0}", 12);
 		template<typename ... Args>
-		void Trace(std::string_view fmt, Args&& ... args)
+		/**
+ * @brief This function traces a message to the game logger and various other logging systems.
+ * 
+ * The function takes in a format string and variable arguments, formats them into a line of text, then logs this line both to the game logger and any number of additional loggers. It also records the offsets at which each character was logged for later retrieval.
+ * 
+ * @param fmt A std::string_view representing the format string.
+ * @param args Variable arguments representing values to be inserted into the format string.
+ * 
+ * @return void
+ */
+void Trace(std::string_view fmt, Args&& ... args)
 		{
 			std::string Line = "[AGEGAME] " + std::vformat(fmt, std::make_format_args(args...)) + "\n";
 			size_t Size = Line.size();
@@ -220,7 +296,15 @@
 #endif
 		}
 		template<typename ... Args>
-		void Info(std::string_view fmt, Args&& ... args)
+		/**
+ * @brief This function logs informational messages to the game logger and various other logging systems.
+ * 
+ * @param fmt A format string that specifies how subsequent arguments are converted for output.
+ * @param args Arguments following the format string.
+ * 
+ * @return None
+ */
+void Info(std::string_view fmt, Args&& ... args)
 		{
 			std::string Line = "[AGEGAME] " + std::vformat(fmt, std::make_format_args(args...)) + "\n";
 			size_t Size = Line.size();
@@ -241,7 +325,15 @@
 #endif
 		}
 		template<typename ... Args>
-		void Warn(std::string_view fmt, Args&& ... args)
+		/**
+ * @brief This function is used for logging warnings in the game. It takes a format string and variable arguments, formats them into a string, 
+ *        then logs this string as a warning along with other information such as log type and offsets. The formatted string is also added to 
+ *        the game's logs. If the platform is Windows, it uses OutputDebugString for logging; if it's Linux, printf is used instead.
+ * @param fmt A format string that specifies how the remaining arguments are converted to strings.
+ * @param ...args Variable number of arguments, which will be formatted according to the provided format string.
+ * @return void
+ */
+void Warn(std::string_view fmt, Args&& ... args)
 		{
 			std::string Line = "[AGEGAME] " + std::vformat(fmt, std::make_format_args(args...)) + "\n";
 			size_t Size = Line.size();
@@ -262,7 +354,16 @@
 #endif
 		}
 		template<typename ... Args>
-		void Error(std::string_view fmt, Args&& ... args)
+		/**
+ * @brief This function is used for logging errors in the game. 
+ * It takes a format string and variable arguments, formats them into a line of text, logs it to various outputs (game log, debug output on Windows, standard output on Linux), and stores it for later retrieval.
+ *
+ * @param fmt The format string used to generate the error message. This should be a printf-style format string.
+ * @param args The variable arguments that will replace placeholders in the format string. 
+ *
+ * @return void No return value is expected as this function only logs errors and does not handle them.
+ */
+void Error(std::string_view fmt, Args&& ... args)
 		{
 			std::string Line = "[AGEGAME] " + std::vformat(fmt, std::make_format_args(args...)) + "\n";
 			size_t Size = Line.size();
@@ -283,7 +384,8 @@
 #endif
 		}
 		template<typename ... Args>
-		void Critical(std::string_view fmt, Args&& ... args)
+		
+void Critical(std::string_view fmt, Args&& ... args)
 		{
 			std::string Line = "[AGEGAME] " + std::vformat(fmt, std::make_format_args(args...)) + "\n";
 			size_t Size = Line.size();
@@ -305,7 +407,8 @@
 		}
 #ifdef AGE_ENABLE_ASSERTS
 		template<typename ... Args>
-		void Assert(bool Condition, std::string_view fmt, Args&& ... args)
+		"This function is used for asserting conditions in the code. If a condition fails, it logs a critical message and triggers a debug breakpoint."
+void Assert(bool Condition, std::string_view fmt, Args&& ... args)
 		{
 			if (!(Condition)) {
 				std::string Line = "[AGEGAMEASSERT] " + std::vformat(fmt, std::make_format_args(args...)) + "\n";
@@ -335,6 +438,13 @@
 		}
 #else //Just Do Nothing
 		template<typename ... Args>
+/**
+ * @brief This function is used to assert a condition. If the condition is not met, it throws an exception with a formatted message.
+ * 
+ * @param Condition The condition that needs to be checked.
+ * @param fmt A format string for the error message.
+ * @param args Arguments for the format string.
+ */
 void Assert(bool Condition, std::string_view fmt, Args&& ... args){}
 #endif
 	}
