--- a/C:\AGE\AGE2D\Engine\src\AGE\Core\Public\Window.h
+++ b/C:\AGE\AGE2D\Engine\src\AGE\Core\Public\Window.h
@@ -18,7 +18,19 @@
 		unsigned int Height;
 		const char* String;
 
-		WindowProps(const std::string& T = "Alcoy Game Engine Editor", unsigned int W = 1280, unsigned int H = 720, const char* S = "")
+		/**
+ * @brief Constructs a WindowProps object with default values.
+ * 
+ * The function initializes the properties of a window, including its title, width, height and string. If no arguments are provided, it uses default values.
+ * 
+ * @param T A string representing the title of the window (default: "Alcoy Game Engine Editor").
+ * @param W An unsigned integer representing the width of the window in pixels (default: 1280).
+ * @param H An unsigned integer representing the height of the window in pixels (default: 720).
+ * @param S A C-style string representing additional information about the window (default: "").
+ * 
+ * @return A WindowProps object with properties set according to the provided arguments. If no arguments are provided, it uses default values.
+ */
+WindowProps(const std::string& T = "Alcoy Game Engine Editor", unsigned int W = 1280, unsigned int H = 720, const char* S = "")
 			: Title(T), Width(W), Height(H), String(S)
 		{
 
@@ -27,13 +39,19 @@
 
 	//Represents a desktop system based Window
 
-	class AGE_API AGEWindow
+	
+class AGE_API AGEWindow
 	{
 	public:
 		
 		using EventCallbackFn = std::function<void(Event&)>;
 
-		virtual ~AGEWindow() {}
+		/**
+ * @brief Virtual destructor for the AGEWindow class.
+ *
+ * This function is responsible for releasing any resources that were acquired by the object during its lifetime. It does not return anything and has no parameters.
+ */
+virtual ~AGEWindow() {}
 
 		virtual void OnUpdate() = 0;
 
