--- a/C:\AGE\AGE2D\Engine\src\AGE\Serializers\Private\IniWriter.cpp
+++ b/C:\AGE\AGE2D\Engine\src\AGE\Serializers\Private\IniWriter.cpp
@@ -9,7 +9,12 @@
 
 namespace AGE
 {
-	IniWriter::IniWriter(const std::filesystem::path &Path)
+	/**
+ * @brief Constructs an IniWriter object with a given path to the INI file.
+ * @param Path The path to the INI file.
+ * @return None
+ */
+IniWriter::IniWriter(const std::filesystem::path &Path)
 		:m_IniPath(Path)
 	{
 		m_Ini.SetUnicode();
@@ -22,14 +27,32 @@
 		}
 	}
 
-	bool IniWriter::Write(const std::string &Section, const std::string &Key, const std::string &Value)
+	/**
+ * @brief Writes a key-value pair to the INI file.
+ * 
+ * This function writes a given value for a specific key in a specified section of an INI file. The operation is successful if the key was not present and has been inserted, or if the key was already present and has been updated.
+ * 
+ * @param Section The name of the section to write into.
+ * @param Key The key to be written.
+ * @param Value The value to be associated with the given key.
+ * 
+ * @return True if the operation is successful (key was inserted or updated), false otherwise.
+ */
+bool IniWriter::Write(const std::string &Section, const std::string &Key, const std::string &Value)
 	{
 		SI_Error rc = m_Ini.SetValue(Section.c_str(), Key.c_str(), Value.c_str());
 
 		return rc == SI_INSERTED || rc == SI_UPDATED;
 	}
 
-	bool IniWriter::SaveFile()
+	/**
+ * @brief Saves the INI file with the given path and content.
+ *
+ * This function saves the current INI data to a file at the specified path. It uses the SaveFile method of the m_Ini object, which writes the data to the file system. The function returns true if the save operation was successful (indicated by SI_OK return code), and false otherwise.
+ *
+ * @return True if the INI file was successfully saved, False otherwise.
+ */
+bool IniWriter::SaveFile()
 	{
 		SI_Error rc = m_Ini.SaveFile(m_IniPath.c_str());
 
