--- a/C:\AGE\AGE2D\Engine\src\AGE\Render\Public\Pipeline.h
+++ b/C:\AGE\AGE2D\Engine\src\AGE\Render\Public\Pipeline.h
@@ -84,7 +84,15 @@
 
 
 		std::unordered_map<std::string, Ref<VertexBuffer>> VertexBuffers;
-		Ref<VertexBuffer> GetVertexBuffer(const std::string& Name)
+		/**
+ * @brief Retrieves a vertex buffer by its name.
+ *
+ * This function takes in the name of a vertex buffer and returns it if it exists. If no such buffer exists, an empty reference is returned.
+ *
+ * @param Name The name of the vertex buffer to retrieve.
+ * @return A reference to the requested vertex buffer or an empty reference if no such buffer exists.
+ */
+Ref<VertexBuffer> GetVertexBuffer(const std::string& Name)
 		{
 			return VertexBuffers[Name]; 
 		}
@@ -97,7 +105,12 @@
 	{
 	public:
 
-		virtual ~Pipeline() = default;
+		/**
+ * @brief Virtual destructor for the Pipeline class.
+ *
+ * This function is responsible for releasing any resources that were acquired by the object during its lifetime. It does not return anything and thus, it has a void return type. 
+ */
+virtual ~Pipeline() = default;
 
 		virtual void Init() = 0;
 		virtual void StartBatch2D() = 0;
