--- a/C:\AGE\AGE2D\Engine\src\AGE\Render\Public\Renderer.h
+++ b/C:\AGE\AGE2D\Engine\src\AGE\Render\Public\Renderer.h
@@ -22,8 +22,20 @@
 
 			static void Flush();
 
-			static inline RendererAPI::API GetAPI() { return RendererAPI::GetAPI(); }
-			static inline void SetAPI(RendererAPI::API Renderer) { RendererAPI::SetAPI(Renderer); }
+			/**
+ * @brief This function returns the current API being used by the renderer.
+ * @return The enum value representing the currently active API (RendererAPI::OpenGL, RendererAPI::Vulkan, etc.).
+ */
+static inline RendererAPI::API GetAPI() { return RendererAPI::GetAPI(); }
+			/**
+ * @brief Sets the rendering API to be used by the application.
+ *
+ * This function sets the API that will be used for rendering operations in the application. The API can be one of three types: OpenGL, DirectX or Vulkan. 
+ * It is important to note that this function does not handle any errors or exceptions related to invalid APIs. Therefore, it's crucial to ensure that the input value is a valid one (OpenGL, DirectX or Vulkan).
+ *
+ * @param Renderer The API to be set for rendering operations. It can be either RendererAPI::API::OPENGL, RendererAPI::API::DIRECTX or RendererAPI::API::VULKAN.
+ */
+static inline void SetAPI(RendererAPI::API Renderer) { RendererAPI::SetAPI(Renderer); }
 		private:
 			struct SceneData
 			{
