--- a/C:\AGE\AGE2D\Engine\src\AGE\Render\Public\Shader.h
+++ b/C:\AGE\AGE2D\Engine\src\AGE\Render\Public\Shader.h
@@ -12,7 +12,12 @@
 	class Shader
 	{
 	public:
-		virtual ~Shader() {};
+		/**
+ * @brief Virtual destructor for the Shader class.
+ *
+ * This function is responsible for freeing any resources that were allocated during the lifetime of the object, such as GPU memory or shaders. It does not return anything and thus has an empty return type (void).
+ */
+virtual ~Shader() {};
 
 		virtual void Bind() const = 0;
 		virtual void Unbind() const = 0;
@@ -51,7 +56,14 @@
 
 		bool Exists(const std::string& Name);
 
-		std::unordered_map<std::string, Ref<Shader>> GetLibrary() { return m_Shaders; }
+		/**
+ * @brief Returns a reference to the shader library.
+ *
+ * This function returns an unordered map where keys are strings and values are references to Shader objects. The purpose of this function is to provide access to all currently loaded shaders in the system. 
+ *
+ * @return An unordered_map containing string-Shader pairs representing the names and corresponding shaders of all loaded shaders.
+ */
+std::unordered_map<std::string, Ref<Shader>> GetLibrary() { return m_Shaders; }
 
 	private:
 		std::unordered_map<std::string, Ref<Shader>> m_Shaders;
