--- a/C:\AGE\AGE2D\Engine\src\AGE\Render\Private\FrameBuffer.cpp
+++ b/C:\AGE\AGE2D\Engine\src\AGE\Render\Private\FrameBuffer.cpp
@@ -5,7 +5,14 @@
 
 namespace AGE
 {
-	Ref<FrameBuffer> FrameBuffer::Create(const FrameBufferSpecification& Spec)
+	/**
+ * @brief Creates a new FrameBuffer based on the specified specification.
+ * The type of FrameBuffer to create is determined by the current RendererAPI in use.
+ * 
+ * @param Spec The specification for the FrameBuffer to be created. This includes things like width, height and color attachments.
+ * @return A reference to the newly created FrameBuffer. If the RendererAPI in use does not support a FrameBuffer of that type, nullptr is returned instead.
+ */
+Ref<FrameBuffer> FrameBuffer::Create(const FrameBufferSpecification& Spec)
 	{
 			switch (Renderer::GetAPI())
 			{
@@ -26,7 +33,13 @@
 	}
 
 	template<typename T>
-	T* FrameBuffer::As()
+	/**
+ * @brief This function is currently not implemented and will always throw an assertion.
+ * It returns a null pointer of type T*. The purpose of this function is unknown.
+ * 
+ * @return A null pointer of type T*
+ */
+T* FrameBuffer::As()
 	{
 		CoreLogger::Assert(false, "As() Failed!");
 		return nullptr;
