--- a/C:\AGE\AGE2D\Engine\src\AGE\Render\Private\RenderBuffer.cpp
+++ b/C:\AGE\AGE2D\Engine\src\AGE\Render\Private\RenderBuffer.cpp
@@ -5,7 +5,8 @@
 
 namespace AGE
 {
-	Ref<VertexBuffer> VertexBuffer::Create(Matrix3D* Vertices, uint32_t Size)
+	
+Ref<VertexBuffer> VertexBuffer::Create(Matrix3D* Vertices, uint32_t Size)
 	{
 		switch (Renderer::GetAPI())
 		{
@@ -25,7 +26,13 @@
 		return nullptr;
 	}
 
-	Ref<VertexBuffer> VertexBuffer::Create(uint32_t Size)
+	/**
+ * @brief Creates a new VertexBuffer of the specified size. The type and usage of the buffer are determined by the current Rendering API in use.
+ * 
+ * @param Size The size of the vertex buffer to create, in bytes.
+ * @return A reference to the newly created VertexBuffer.
+ */
+Ref<VertexBuffer> VertexBuffer::Create(uint32_t Size)
 	{
 		switch (Renderer::GetAPI())
 		{
@@ -45,7 +52,12 @@
 		return nullptr;
 	}
 
-	Ref<VertexBuffer> VertexBuffer::Create(float* Vertices, uint32_t Size)
+	/**
+ * @brief Creates a new vertex buffer object.
+ * 
+ * This function creates and initializes a new VertexBuffer object based on the current rendering API in use. The data for the vertices is passed as an array of floats, with the size of this array also provided. If an unsupported rendering API is currently in use, nullptr is returned.
+ */
+Ref<VertexBuffer> VertexBuffer::Create(float* Vertices, uint32_t Size)
 	{
 		switch (Renderer::GetAPI())
 		{
@@ -66,7 +78,8 @@
 	}
 
 
-	Ref<IndexBuffer> IndexBuffer::Create(uint32_t* Indices, uint32_t Count)
+	
+Ref<IndexBuffer> IndexBuffer::Create(uint32_t* Indices, uint32_t Count)
 	{
 		switch (Renderer::GetAPI())
 		{
@@ -85,10 +98,25 @@
 		CoreLogger::Assert(false, "Unknown Renderer API!");
 		return nullptr;
 	}
-	BufferLayout::BufferLayout()
+	/**
+ * @brief BufferLayout is a class that represents the layout of a buffer in memory. 
+ * It provides methods for adding elements to the buffer and retrieving them by index.
+ * The buffer can hold any type of data, but it's typically used with primitive types like int, float, etc.
+ */
+BufferLayout::BufferLayout()
 	{
 	}
-	Ref<UniformBuffer> UniformBuffer::Create(uint32_t Size, uint32_t Binding)
+	/**
+ * @brief Creates a new UniformBuffer instance.
+ * 
+ * This function creates and initializes a new UniformBuffer object of the specified size and binding point. The type of buffer to be created is determined by the current Renderer API in use. If OpenGL is currently being used, an OpenGLUniformBuffer will be created; otherwise, if no supported API is found, null is returned.
+ * 
+ * @param Size The size (in bytes) of the UniformBuffer to create.
+ * @param Binding The binding point for the UniformBuffer in the shader program.
+ * 
+ * @return A reference to a new UniformBuffer object if successful, nullptr otherwise.
+ */
+Ref<UniformBuffer> UniformBuffer::Create(uint32_t Size, uint32_t Binding)
 	{
 		switch (Renderer::GetAPI())
 		{
@@ -105,19 +133,37 @@
 	}
 
 	template<typename T>
-	T* VertexBuffer::As()
+	/**
+ * @brief This function is currently not implemented and will always throw an assertion.
+ * It returns a null pointer of type T*. The purpose of this function is unknown.
+ * 
+ * @return Returns a null pointer of type T*
+ */
+T* VertexBuffer::As()
 	{
 		CoreLogger::Assert(false, "As() Failed!");
 		return nullptr;
 	}
 	template<typename T>
-	T* IndexBuffer::As()
+	/**
+ * @brief This function is currently not implemented and will always return a null pointer.
+ * It's intended to provide the ability to cast this IndexBuffer instance to another type, but it's not yet supported. 
+ * The function will assert false with an error message indicating that As() Failed!
+ * @return nullptr Always returns nullptr.
+ */
+T* IndexBuffer::As()
 	{
 		CoreLogger::Assert(false, "As() Failed!");
 		return nullptr;
 	}
 	template<typename T>
-	T* UniformBuffer::As()
+	/**
+ * @brief This function is currently not implemented and will always throw an assertion.
+ * It returns a null pointer of type T*. The purpose of this function is unknown.
+ * 
+ * @return A null pointer of type T*
+ */
+T* UniformBuffer::As()
 	{
 		CoreLogger::Assert(false, "As() Failed!");
 		return nullptr;
