--- a/C:\AGE\AGE2D\Engine\src\AGE\Render\Private\VertexArray.cpp
+++ b/C:\AGE\AGE2D\Engine\src\AGE\Render\Private\VertexArray.cpp
@@ -5,7 +5,14 @@
 
 namespace AGE
 {
-	Ref<VertexArray> VertexArray::Create()
+	/**
+ * @brief Creates a new VertexArray object based on the current RendererAPI.
+ * 
+ * This function creates and returns a reference to a new VertexArray object, which is specific to the currently used RendererAPI. If the API is not supported or unknown, it asserts false and returns nullptr.
+ * 
+ * @return Ref<VertexArray> A reference to the newly created VertexArray object.
+ */
+Ref<VertexArray> VertexArray::Create()
 	{
 		switch (Renderer::GetAPI())
 		{
