--- a/C:\AGE\AGE2D\Engine\src\AGE\Parser\Public\FbxParser.h
+++ b/C:\AGE\AGE2D\Engine\src\AGE\Parser\Public\FbxParser.h
@@ -9,9 +9,20 @@
 	class FBXParser
 	{
 	public:
-		FBXParser() = default;
+		/**
+ * @brief Default constructor for the FBXParser class.
+ *
+ * This function initializes an instance of the FBXParser class with default values. It does not perform any specific operations or require any parameters to be set. 
+ *
+ * @return A new instance of the FBXParser class with all fields initialized to their default values.
+ */
+FBXParser() = default;
 
-		static FBXParser& Get()
+		/**
+ * @brief This function returns a reference to the singleton instance of the FBXParser class. If an instance does not already exist, it will be created.
+ * @return A reference to the single instance of the FBXParser class.
+ */
+static FBXParser& Get()
 		{
 			if (!s_Instance)
 			{
