--- a/C:\AGE\AGE2D\Engine\src\AGE\VisualScripting\Public\NodeEditorWindow.h
+++ b/C:\AGE\AGE2D\Engine\src\AGE\VisualScripting\Public\NodeEditorWindow.h
@@ -26,9 +26,23 @@
 		void OnAttach();
 		void OnImGuiRender(TimeStep DeltaTime);
 
-		const std::string& GetWindowName() const { return m_Name; }
-
-		static void Serialize(DataWriter* Serializer, const NodeEditorWindow& Data)
+		/**
+ * @brief Returns the name of the window.
+ * @return A constant reference to a string containing the name of the window.
+ */
+const std::string& GetWindowName() const { return m_Name; }
+
+		/**
+ * @brief This function serializes the NodeEditorWindow data into a DataWriter object.
+ * 
+ * The function takes in a pointer to a DataWriter and a constant reference to a NodeEditorWindow. It writes out various properties of the NodeEditorWindow, including its name, ID, nodes, links, and whether it shows ordinals or not.
+ * 
+ * @param Serializer A pointer to the DataWriter object that will be used for serialization.
+ * @param Data The constant reference to the NodeEditorWindow whose data is being serialized.
+ * 
+ * @return void
+ */
+static void Serialize(DataWriter* Serializer, const NodeEditorWindow& Data)
 		{
 			std::vector<AGENode> Nodes(Data.m_Nodes.size());
 			std::vector<AGENodeLink> Links(Data.m_Links.size());
@@ -48,7 +62,8 @@
 			Serializer->WriteRaw<bool>(Data.m_ShowOrdinals);
 		}
 
-		static void Deserialize(DataReader* Serializer, NodeEditorWindow& Data)
+		
+static void Deserialize(DataReader* Serializer, NodeEditorWindow& Data)
 		{
 			uint64_t ID;
 			std::vector<AGENode> Nodes;
