--- a/C:\AGE\AGE2D\Engine\src\AGE\UI\Components\Public\ButtonComponent.h
+++ b/C:\AGE\AGE2D\Engine\src\AGE\UI\Components\Public\ButtonComponent.h
@@ -19,15 +19,36 @@
 		void DrawContent() override;
 		void OnUpdate(TimeStep DeltaTime) override;
 		void OnEvent(Event& Event) override;
-		void CallSerialize(DataWriter* Serializer) override
+		/**
+ * @brief This function is used to serialize the VerticalBoxComponent object.
+ * 
+ * @param[in] Serializer A pointer to a DataWriter object, which provides methods for writing data.
+ * @return void No return value expected.
+ */
+void CallSerialize(DataWriter* Serializer) override
 		{
 			//Serializer->WriteObject<VerticalBoxComponent>(*this);
 		}
-		void CallDeserialize(DataReader* Serializer) override
+		/**
+ * @brief This function is used to deserialize data from a DataReader object. 
+ * It reads an object of type VerticalBoxComponent using the provided serializer.
+ *
+ * @param Serializer A pointer to the DataReader object that will be used for deserialization.
+ */
+void CallDeserialize(DataReader* Serializer) override
 		{
 			//Serializer->ReadObject<VerticalBoxComponent>(*this);
 		}
-		void SetOnClickFunc(const std::function<void()>& func) { m_OnClick = func; }
+		/**
+ * @brief Sets the on-click function for this object.
+ *
+ * This function sets a new function to be called when the object is clicked. The provided function should take no arguments and return void.
+ *
+ * @param func A std::function that will be set as the new click handler. It should have no parameters and return nothing.
+ * 
+ * @return None
+ */
+void SetOnClickFunc(const std::function<void()>& func) { m_OnClick = func; }
 
 		RTTR_ENABLE(UIComponent)
 		RTTR_REGISTRATION_FRIEND
