--- a/C:\AGE\AGE2D\Engine\src\AGE\UI\Components\Public\TextBoxComponent.h
+++ b/C:\AGE\AGE2D\Engine\src\AGE\UI\Components\Public\TextBoxComponent.h
@@ -27,16 +27,39 @@
 
 		void DrawFontSelectionComboBox() override;
 		void DrawContent() override;
-		void CallSerialize(DataWriter* Serializer) override
+		/**
+ * @brief This function serializes the TextBoxComponent object using a DataWriter.
+ * 
+ * @param[in] Serializer A pointer to an instance of DataWriter that is used for writing data.
+ */
+void CallSerialize(DataWriter* Serializer) override
 		{
 			Serializer->WriteObject<TextBoxComponent>(*this);
 		}
-		void CallDeserialize(DataReader* Serializer) override
+		/**
+ * @brief Deserializes the TextBoxComponent from a DataReader.
+ * 
+ * This function reads an instance of TextBoxComponent from the provided DataReader and assigns it to this object.
+ * The exact format in which the data is read depends on how the DataReader's ReadObject method is implemented.
+ *
+ * @param Serializer A pointer to a DataReader that provides the serialized data.
+ */
+void CallDeserialize(DataReader* Serializer) override
 		{
 			Serializer->ReadObject<TextBoxComponent>(*this);
 		}
 
-		static void Serialize(DataWriter* Serializer, const TextBoxComponent& Instance)
+		/**
+ * @brief Serializes the TextBoxComponent instance into a DataWriter object.
+ * 
+ * This function writes various properties of the TextBoxComponent to the provided DataWriter, including its name, type, string properties (text, font name, color and size), and position and rotation data.
+ * 
+ * @param Serializer Pointer to the DataWriter instance where serialization will be performed.
+ * @param Instance The TextBoxComponent instance that needs to be serialized.
+ * 
+ * @return void
+ */
+static void Serialize(DataWriter* Serializer, const TextBoxComponent& Instance)
 		{
 			Serializer->WriteString(Instance.m_Name);
 			Serializer->WriteRaw<uint16_t>(Instance.m_Type);
@@ -55,7 +78,15 @@
 			Serializer->WriteRaw<float>(Instance.m_StringProperties.Rotation.z);
 		}
 
-		static void Deserialize(DataReader* Serializer, TextBoxComponent& Instance)
+		/**
+ * @brief Deserializes the TextBoxComponent instance from a DataReader object.
+ * 
+ * This function reads various properties of the TextBoxComponent instance, including its name, type, string properties (text, font name, color and size), and position and rotation data. It uses the provided Serializer to read these values.
+ * 
+ * @param[in] Serializer A pointer to a DataReader object that provides the serialized data.
+ * @param[out] Instance The TextBoxComponent instance to be deserialized.
+ */
+static void Deserialize(DataReader* Serializer, TextBoxComponent& Instance)
 		{
 			Serializer->ReadString(Instance.m_Name);
 			uint16_t Type;
